django/template/base.py

```diff
@@ -633,19 +633,19 @@ def find_filter(self, filter_name):
 
 filter_raw_string = r"""
 ^(?P<constant>%(constant)s)|
-^(?P<var>[%(var_chars)s]+|%(num)s)|
+^(?P<var>%(num)s$|[%(var_chars)s]+)|
  (?:\s*%(filter_sep)s\s*
      (?P<filter_name>\w+)
          (?:%(arg_sep)s
              (?:
               (?P<constant_arg>%(constant)s)|
-              (?P<var_arg>[%(var_chars)s]+|%(num)s)
+              (?P<var_arg>%(num)s$|[%(var_chars)s]+)
```

Comment:
I can revert these two changes without any test failures - are they needed?

Reply:
By adding '-' to `var_chars`, that original test case ('5.2e-3') passes with or without any regex changes. BUT it goes back to the original state that all "nums" other than those containing a '+' are matching against `var_chars` rather than `num`. If we don't care about which group they match with and we're not capturing them separately, maybe we should just add '+' to `var_chars` and only check against that pattern?

Reply:
That change I like 👍 I mean these two changes where we have swapped num and var_chars. I can revert those 2 line changes without any test failures